src/ILLink.Shared/XmlProcessorBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Xml.XPath;
+
+namespace ILLink.Shared
+{
+	public abstract class XmlProcessorBase
```

Comment:
Yes, since ProcessLinkerXml is a class and NodeBase is a record and ideally we will share the NodeBase implementation in the linker, I made it base of NodeBase.